include/simdjson/inline/document.h

```diff
@@ -1161,13 +1202,46 @@ really_inline T tape_ref::next_tape_value() const noexcept {
   memcpy(&x,&doc->tape[json_index + 1],sizeof(uint64_t));
   return x;
 }
+
+ /**
+ * Design notes:
+ * Instead of having references to the low-level string_buf all over the
+ * code, we hide all of the logic in get_string_length and get_c_str
+ * and build everything from there. Note that get_string_length is
+ * expected to be a very fast function, so it should be broadly available
+ */
+
+really_inline uint32_t internal::tape_ref::get_string_length() const noexcept {
+  uint64_t tape_val = uint64_t(tape_value());
+  uint32_t len = (tape_val >> 32) & 0xFFFFFF; // grab 3 bytes
+  if(unlikely(len > 0x7fffff)) {  // only go there if the highest bit is set
+    uint32_t string_buf_index = uint32_t(tape_val); // we will need to touch the string buffer
+    // we have a long string and we need to jump through some loops
+    len ^= 0x800000; // set the high bit to zero
```

Comment:
`0x800000 << 9 == 0` // this can't be right... should it be a shift by 8?

Reply:
Good catch. Let me fix this.

Reply:
Note that the code is correct, but the xor is useless (for the reason you point out).